Make BaseCache an ABC (#683)
aio-libs/aiocache: CHANGES.rst

* The ``key`` parameter has been removed from the ``cached`` decorator. The behaviour can be easily reimplemented with ``key_builder=lambda *a, **kw: "foo"``
* When using the ``key_builder`` parameter in ``@multicached``, the function will now return the original, unmodified keys, only using the transformed keys in the cache (this has always been the documented behaviour, but not the implemented behaviour).

> * ``BaseSerializer`` is now an ``ABC``, so cannot be instantiated directly.
> * ``BaseCache`` is now an ``ABC``, so cannot be instantiated directly.

Reviewer: ```suggestion
* ``BaseCache`` and ``BaseSerializer`` are now ``ABC``s, so cannot be instantiated directly.
```
